hijack/middleware.py

```diff
@@ -13,18 +15,22 @@
 class HijackUserMiddleware(MiddlewareMixin):
     """Set `is_hijacked` attribute; render and inject notification."""
 
+    @staticmethod
+    def setup_user(request, user):
+        """Set `is_hijacked` attribute."""
+        user.is_hijacked = bool(request.session.get("hijack_history", []))
+        return user
+
     def process_request(self, request):
         """Set `is_hijacked` and override REMOTE_USER header."""
-        if request.session.is_empty():
-            # do not touch empty sessions to avoid unnecessary vary on cookie header
-            return
```

Comment:
This part is completely gone, so instead of checking for empty session in the request, we will lazily check-set  `is_hijacked` attribute. It'd be still a good optimization, or I am missing something, and it does not make any sense any more?

Reply:
Yes, we don't need that check, if no one is accessing the attribute or the user, we don't need to do anything.